python3Packages.weasyprint: 52 -> 53.4

Prillan opened the pull request:
###### Motivation for this change

ZHF: #144627

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Notes

I have not tested this extensively, but the basic functionality in https://doc.courtbouillon.org/weasyprint/stable/first_steps.html works well.

Prillan committed:
python3Packages.pydyf: init at 1.0.2

Prillan committed:
python3Packages.weasyprint: 52 -> 54.3

risicle commented on pkgs/development/python-modules/pydyf/default.nix:
```diff
@@ -0,0 +1,48 @@
+  # setup.py is auto-generated and doesn't pass the flake8 check
+  checkPhase = ''
+    runHook preCheck
+    pytest --ignore=setup.py
```
This could probably use `pytestCheckHook`: https://nixos.org/manual/nixpkgs/stable/#using-pytestcheckhook

Prillan replied:
Thanks, updated in both `pydyf` and `weasyprint`!

risicle commented:
Nice. `nixpkgs-review` happy, macos 10.15, nixos x86_64 & aarch64.

risicle commented:
I can't explain why it's happening now but wasn't before:

```______________________________ test_font_stretch _______________________________

    @assert_no_logs
    def test_font_stretch():
        page, = render_pages('''
          <style>
            p { float: left; font-family: %s }
          </style>
          <p>Hello, world!</p>
          <p style="font-stretch: condensed">Hello, world!</p>
        ''' % SANS_FONTS)
        html, = page.children
        body, = html.children
        p_1, p_2 = body.children
        normal = p_1.width
        condensed = p_2.width
>       assert condensed < normal
E       assert 101.1953125 < 101.1953125

/build/weasyprint-53.4/tests/layout/test_inline.py:654: AssertionError
```

`python38Packages.weasyprint` on nixos x86_64.

Prillan commented:
> I can't explain why it's happening now but wasn't before:
> 
> ```
> 
>     @assert_no_logs
>     def test_font_stretch():
>         page, = render_pages('''
>           <style>
>             p { float: left; font-family: %s }
>           </style>
>           <p>Hello, world!</p>
>           <p style="font-stretch: condensed">Hello, world!</p>
>         ''' % SANS_FONTS)
>         html, = page.children
>         body, = html.children
>         p_1, p_2 = body.children
>         normal = p_1.width
>         condensed = p_2.width
> >       assert condensed < normal
> E       assert 101.1953125 < 101.1953125
> 
> /build/weasyprint-53.4/tests/layout/test_inline.py:654: AssertionError
> ```
> 
> `python38Packages.weasyprint` on nixos x86_64.
> 
> Edit: and macos 10.15

I removed the `-k "not test_font_stretch"` part of the pytest invocation when I added your suggestion (it worked on my NixOS laptop). I've reverted it now.

Thank you for being thorough! ^^

risicle commented:
Feel like adding yourself as a "maintainer" of either package? It really just means you get pinged when someone makes an update to it, it's not a big responsibility. https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers

risicle commented:
`nixpkgs-review` happy, macos 10.15, nixos x86_64 & aarch64

Prillan committed:
maintainers: add rprecenth

thiagokokada commented:
Result of `nixpkgs-review pr 146315` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>python38Packages.dnachisel</li>
    <li>python38Packages.flametree</li>
    <li>python38Packages.pydyf</li>
    <li>python38Packages.weasyprint</li>
    <li>python38Packages.xml2rfc</li>
    <li>python39Packages.dnachisel</li>
    <li>python39Packages.flametree</li>
    <li>python39Packages.pydyf</li>
    <li>python39Packages.weasyprint</li>
    <li>xml2rfc (python39Packages.xml2rfc)</li>
    <li>sourcehut.metasrht</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/development/python-modules/pydyf/default.nix:
```diff
@@ -0,0 +1,48 @@
+    coverage
+    ghostscript
+    pillow
+    pytest
+    pytest-cov
+    pytest-flake8
+    pytest-isort
```
pytest should not be added to checkInputs when using pytestCheckHook. in pythonPackages we never want to generate cov reports because it just takes time and there is no advantage in doing it. flake8 tests are not run because they easily break on upgrades and don't provide any benefit. Doing a PR with those changes right now.

Prillan replied:
Makes sense, thanks!
